course6/toyapp_mikhail/displaylib/vector_common.h

```diff
@@ -0,0 +1,100 @@
+#pragma once
+
+#include <array>
+
+namespace Display {
+
+template <class TValue, size_t TDimentions, class TDecoratedType = TValue>
```

Comment:
the "T" in "TDimensions" doesn't styand for "type"? right?
The popular convention is to use ALL_CAPS for value template arguments. Just like for constants; they are compile-time constants after all.

Reply:
Why is "TDecoratedType"? It's not imemdiately obvious for mew why would you need that in a vector, which means it needs eithjer better name, or at least explanation in a comment.

Reply:
Ah! Decorated type is the implementation! You are doing CRTP! Then using TValue as default makes no sense, and TSelf is useless.

Reply:
namespace Display::Detail